### PR TITLE
docs: add "Supported pandas APIs" reference to the documentation

### PR DESCRIPTION
**Pull request by tswast**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigquery-dataframes/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

🦕

### CODE REVIEW

**Comment by tswast** [comment]
Preview from a local `nox -r -s docs` build:

Screenshot 2024-03-28 at 5 32 33 PM

**Comment by ashleyxuu** [comment]
The content looks good to me, but I'm not able to generate the page directly through `nox- s docfx`. Let me check with CloudRAD team to see if they have any solution.

**Review by ashleyxuu — COMMENTED** [review]
The content looks good to me, but I'm not able to generate the page directly through `nox -s docfx`. Let me check with CloudRAD team to see if they have any solution.

File: docs/templates/toc.yml [hunk 1]
```diff
@@ -72,6 +72,8 @@
       name: Series
     - name: Window
       uid: bigframes.core.window.Window
+    - name: Supported pandas APIs
+      href: supported_pandas_apis.html
```
Comment (tswast) [inline docs/templates/toc.yml:76]: @ashleyxuu Is this the right way to do this? I can't figure out how to get the HTML file to stage.

Comment (ashleyxuu) [reply docs/templates/toc.yml:76]:
```
- href: supported_pandas_apis.html
      name: Supported pandas APIs

Comment (tswast) [reply docs/templates/toc.yml]: Done.

**Review by ashleyxuu — APPROVED** [review]

**Pull request merged**